src/components/annotations/index.js

```diff
@@ -476,8 +487,15 @@ annotations.draw = function(gd, index, opt, value) {
         // make sure the arrowhead (if there is one)
         // and the annotation center are visible
         if(options.showarrow) {
-            arrowX = Lib.constrain(annPosPx.x - options.ax, 1, fullLayout.width - 1);
-            arrowY = Lib.constrain(annPosPx.y - options.ay, 1, fullLayout.height - 1);
+            if(options.axref === options.xref)
+                arrowX = Lib.constrain(annPosPx.x, 1, fullLayout.width - 1);
+            else
```

Comment:
It should. By we haven't decided on a [brace-style](http://eslint.org/docs/rules/brace-style) yet. Making this rule impossible to implement with eslint at the moment.

Personally, I like the Stroustrup style, but most JS folks like the _one-trye-brace-style_. Moreover, eslint does not have `--fix` support for that rule yet. So implementing this rule would require some (maybe a lot) of line change.